components/server/src/main/java/com/hotels/styx/server/netty/connectors/HttpResponseWriter.java

```diff
@@ -157,7 +156,12 @@ private void onWriteEmptyLastChunkOutcome(ChannelFuture writeOp) {
             return future;
         } catch (Throwable cause) {
             LOGGER.warn("Failed to convert response headers. response={}, Cause={}", new Object[]{response, cause});
-            toObservable(response.body()).forEach(it -> Buffers.toByteBuf(it).release());
+            response.body().subscribe(new BaseSubscriber<Buffer>() {
+                @Override
+                protected void hookOnNext(Buffer b) {
+                    Buffers.toByteBuf(b).release();
+                }
+            });
```

Comment:
I think we should use the `ByteStream.drop` instead:

```
            Flux.from(response.body()
                    .drop())
                    .subscribe();
```

Reply:
Yes, much neater! done :)